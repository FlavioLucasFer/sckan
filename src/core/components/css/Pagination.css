.active-page {
	background-color: #00bfa5 !important;
}