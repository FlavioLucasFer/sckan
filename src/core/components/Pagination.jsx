import './css/Pagination.css';
import React, { useState } from 'react';
import PropTypes from 'prop-types';

Pagination.propTypes = {
	pagesQuantity: PropTypes.number.isRequired,
};

function Pagination(props) {
	const {
		pagesQuantity,
	} = props;

	const [activePage, setActivePage] = useState(1);
		
	function renderPages() {
		const listItems = [];

		for (let i = 1; i <= pagesQuantity; i++) {
			listItems.push(
				<li key={i} 
					className={`${activePage === i ? 'active active-page' : 'waves-effect'}`}
					onClick={() => setActivePage(i)}>
					<a>{i}</a>
				</li>
			);
		}

		return listItems;
	}

	function handleClickOnNextPage() {
		setActivePage(activePage + 1);
	}
	
	function handleClickOnPreviousPage() {
		setActivePage(activePage - 1);
	}

	return (
		<ul className="pagination"
			style={{ textAlign: 'center' }}>
			<li className={activePage === 1 ? 'disabled' : 'waves-effect'}
				onClick={activePage > 1 && handleClickOnPreviousPage}>
				<a><i className="material-icons">chevron_left</i></a>
			</li>
			{renderPages()}
			<li className={activePage === pagesQuantity ? 'disabled' : 'waves-effect'}
				onClick={activePage < pagesQuantity && handleClickOnNextPage}>
				<a><i className="material-icons">chevron_right</i></a>
			</li>
		</ul>
	);
}

export default Pagination;
