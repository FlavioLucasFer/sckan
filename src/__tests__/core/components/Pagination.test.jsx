import React from 'react';
import { shallow } from 'enzyme';

import Pagination from 'core/components/Pagination';

describe('Testing Pagination core component', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<Pagination pagesQuantity={10} />
		);

		expect(wrapper.debug()).toMatchSnapshot();
		expect(wrapper.find('li').length).toBe(12);
		expect(wrapper.find('li').at(1).hasClass('active')).toBeTruthy();
		expect(wrapper.find('li').at(1).hasClass('active-page')).toBeTruthy();
		
		wrapper.setProps({
			pagesQuantity: 5,
		});
		
		expect(wrapper.debug()).toMatchSnapshot();
		expect(wrapper.find('li').length).toBe(7);

		expect(wrapper.find({ id: 'goto-previous-page' }).hasClass('disabled')).toBeTruthy();
		expect(wrapper.find({ id: 'goto-next-page' }).hasClass('disabled')).toBeFalsy();
		
		wrapper.setProps({
			pagesQuantity: 1,
		});
		
		expect(wrapper.find({ id: 'goto-previous-page' }).hasClass('disabled')).toBeTruthy();
		expect(wrapper.find({ id: 'goto-next-page' }).hasClass('disabled')).toBeTruthy();
	});

	it('should have only one page active', () => {
		const wrapper = shallow(
			<Pagination pagesQuantity={3} />
		);

		expect(wrapper.find('.active').length).toBe(1);
		expect(wrapper.find('.active-page').length).toBe(1);
	});

	it('should exchange active page when click on other page', () => {
		const wrapper = shallow(
			<Pagination pagesQuantity={6} />
		);

		expect(wrapper.find('li').at(1).hasClass('active')).toBeTruthy();
		expect(wrapper.find('li').at(1).hasClass('active-page')).toBeTruthy();
		
		wrapper.find('li').at(5).simulate('click');
		
		expect(wrapper.find('li').at(1).hasClass('active')).toBeFalsy();
		expect(wrapper.find('li').at(1).hasClass('active-page')).toBeFalsy();
		expect(wrapper.find('li').at(5).hasClass('active')).toBeTruthy();
		expect(wrapper.find('li').at(5).hasClass('active-page')).toBeTruthy();
	});

	it('should exchange active page to next page when click on next page li', () => {
		const wrapper = shallow(
			<Pagination pagesQuantity={8} />
		);

		expect(wrapper.find('li').at(1).hasClass('active')).toBeTruthy();
		expect(wrapper.find('li').at(1).hasClass('active-page')).toBeTruthy();

		wrapper.find({ id: 'goto-next-page' }).simulate('click');

		expect(wrapper.find('li').at(1).hasClass('active')).toBeFalsy();
		expect(wrapper.find('li').at(1).hasClass('active-page')).toBeFalsy();
		expect(wrapper.find('li').at(2).hasClass('active')).toBeTruthy();
		expect(wrapper.find('li').at(2).hasClass('active-page')).toBeTruthy();
	});

	it('should exchange active page to previous page when click on previous page li', () => {
		const wrapper = shallow(
			<Pagination pagesQuantity={8} />
		);

		expect(wrapper.find('li').at(1).hasClass('active')).toBeTruthy();
		expect(wrapper.find('li').at(1).hasClass('active-page')).toBeTruthy();
		
		wrapper.find({ id: 'goto-next-page' }).simulate('click');
		expect(wrapper.find('li').at(1).hasClass('active')).toBeFalsy();
		expect(wrapper.find('li').at(1).hasClass('active-page')).toBeFalsy();
		expect(wrapper.find('li').at(2).hasClass('active')).toBeTruthy();
		expect(wrapper.find('li').at(2).hasClass('active-page')).toBeTruthy();
		
		wrapper.find({ id: 'goto-previous-page' }).simulate('click');
		expect(wrapper.find('li').at(2).hasClass('active')).toBeFalsy();
		expect(wrapper.find('li').at(2).hasClass('active-page')).toBeFalsy();
		expect(wrapper.find('li').at(1).hasClass('active')).toBeTruthy();
		expect(wrapper.find('li').at(1).hasClass('active-page')).toBeTruthy();
	});
});