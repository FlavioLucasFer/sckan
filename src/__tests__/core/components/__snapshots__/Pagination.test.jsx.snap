// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Pagination core component should render correctly 1`] = `
"<ul className=\\"pagination\\" style={{...}}>
  <li id=\\"goto-previous-page\\" className=\\"disabled\\" onClick={false}>
    <a>
      <i className=\\"material-icons\\">
        chevron_left
      </i>
    </a>
  </li>
  <li className=\\"active active-page\\" onClick={[Function: onClick]}>
    <a>
      1
    </a>
  </li>
  <li className=\\"waves-effect\\" onClick={[Function: onClick]}>
    <a>
      2
    </a>
  </li>
  <li className=\\"waves-effect\\" onClick={[Function: onClick]}>
    <a>
      3
    </a>
  </li>
  <li className=\\"waves-effect\\" onClick={[Function: onClick]}>
    <a>
      4
    </a>
  </li>
  <li className=\\"waves-effect\\" onClick={[Function: onClick]}>
    <a>
      5
    </a>
  </li>
  <li className=\\"waves-effect\\" onClick={[Function: onClick]}>
    <a>
      6
    </a>
  </li>
  <li className=\\"waves-effect\\" onClick={[Function: onClick]}>
    <a>
      7
    </a>
  </li>
  <li className=\\"waves-effect\\" onClick={[Function: onClick]}>
    <a>
      8
    </a>
  </li>
  <li className=\\"waves-effect\\" onClick={[Function: onClick]}>
    <a>
      9
    </a>
  </li>
  <li className=\\"waves-effect\\" onClick={[Function: onClick]}>
    <a>
      10
    </a>
  </li>
  <li id=\\"goto-next-page\\" className=\\"waves-effect\\" onClick={[Function: handleClickOnNextPage]}>
    <a>
      <i className=\\"material-icons\\">
        chevron_right
      </i>
    </a>
  </li>
</ul>"
`;

exports[`Testing Pagination core component should render correctly 2`] = `
"<ul className=\\"pagination\\" style={{...}}>
  <li id=\\"goto-previous-page\\" className=\\"disabled\\" onClick={false}>
    <a>
      <i className=\\"material-icons\\">
        chevron_left
      </i>
    </a>
  </li>
  <li className=\\"active active-page\\" onClick={[Function: onClick]}>
    <a>
      1
    </a>
  </li>
  <li className=\\"waves-effect\\" onClick={[Function: onClick]}>
    <a>
      2
    </a>
  </li>
  <li className=\\"waves-effect\\" onClick={[Function: onClick]}>
    <a>
      3
    </a>
  </li>
  <li className=\\"waves-effect\\" onClick={[Function: onClick]}>
    <a>
      4
    </a>
  </li>
  <li className=\\"waves-effect\\" onClick={[Function: onClick]}>
    <a>
      5
    </a>
  </li>
  <li id=\\"goto-next-page\\" className=\\"waves-effect\\" onClick={[Function: handleClickOnNextPage]}>
    <a>
      <i className=\\"material-icons\\">
        chevron_right
      </i>
    </a>
  </li>
</ul>"
`;
